### PR TITLE
Add component based cases

### PR DESCRIPTION
**Pull request by epeli**
First attempt. Don't like at all how the components are detected in the children prop. Ideas?

closes #16

### CODE REVIEW

File: __tests__/component-test.js [hunk 1]
```diff
@@ -0,0 +1,38 @@
...
+		);
+		let val = TestUtils.findRenderedDOMComponentWithTag(component, 'h1');
+		equal(val.getDOMNode().textContent, 'expected');
+	});
+
+  it('should render the default value if no cases match', () => {
+		let component = TestUtils.renderIntoDocument(
+			<Cond value={10}>
+				<Case test={2}><h1>unexpected</h1></Case>
+				<Case test={1}><h1>expected</h1></Case>
```
Comment (stoeffel) [inline __tests__/component-test.js]: Could you add a test with `test={someFunc}`?

**Comment by stoeffel** [comment]
Unfortunately, I have no other ideas :frowning:  That's the way I had done it.

File: __tests__/component-test.js (hunk 1 above)
Comment (stoeffel) [inline __tests__/component-test.js]: and maybe one with mixed clauses.
```jsx
<Case test={2}><h1>unexpected</h1></Case>
{[ 3, <h1>nope</h1> ]}
```

Comment (epeli) [inline __tests__/component-test.js]: Yeah, I was also thinking that they could be easily mixed but I don't think there is any reasonable use case for it.

Comment (stoeffel) [inline __tests__/component-test.js]: We should just state that in the readme. that you can't mix it.

**Comment by epeli** [comment]
> Could you add a test with test={someFunc}?

Added, rebased and force pushed.

File: __tests__/component-test.js (hunk 1 above)
Comment (epeli) [inline __tests__/component-test.js]: Agreed.

**Comment by stoeffel** [comment]
thx a lot. We will just have to add it to the readme and this is good to go. we can always refactor detection of the child components later.

**Comment by stoeffel** [comment]
btw. I added you to npm.

**Comment by epeli** [comment]
> btw. I added you to npm.

Thanks.

Renamed `Case` to `Clause` as it's more inline with the wording in the readme.

**Comment by stoeffel** [comment]
I added some more docs.

**Pull request merged**